plugins/main/bridge/bridge.go

```diff
@@ -127,6 +142,59 @@ func loadNetConf(bytes []byte, envArgs string) (*NetConf, string, error) {
 	return n, n.CNIVersion, nil
 }
 
+// This method is copied from https://github.com/k8snetworkplumbingwg/ovs-cni/blob/v0.27.2/pkg/plugin/plugin.go
+func collectVlanTrunk(vlanTrunk []*VlanTrunk) ([]int, error) {
+	if vlanTrunk == nil {
+		return nil, nil
+	}
+
+	vlanMap := make(map[int]bool)
+	for _, item := range vlanTrunk {
+		var minID int
+		var maxID int
+		var ID int
+		if item.MinID != nil {
+			minID = *item.MinID
+			if minID < 0 || minID > 4094 {
+				return nil, errors.New("incorrect trunk minID parameter")
+			}
+		}
+		if item.MaxID != nil {
+			maxID = *item.MaxID
+			if maxID < 0 || maxID > 4094 {
+				return nil, errors.New("incorrect trunk maxID parameter")
+			}
+			if maxID < minID {
+				return nil, errors.New("minID is greater than maxID in trunk parameter")
+			}
+		}
+		if minID > 0 && maxID > 0 {
+			for v := minID; v <= maxID; v++ {
+				vlanMap[v] = true
+			}
+		}
+
+		// single vid
+		if item.ID != nil {
+			ID = *item.ID
+			if ID < 0 || ID > 4094 {
+				return nil, errors.New("incorrect trunk id parameter")
+			}
+			vlanMap[ID] = true
+		}
+	}
+
+	if len(vlanMap) == 0 {
+		return nil, nil
+	}
+	vlans := make([]int, 0, len(vlanMap))
+	for k := range vlanMap {
+		vlans = append(vlans, k)
+	}
```

Comment:
I may be missing something, but I don't understand why you first compute a set of the VLAN IDs, and then compute a slice from it.

Couldn't you have just created the slice directly ? And append the trunk IDs to it?

Reply:
I copy those from OVS cni.
IIRC, we want to deduplicate vlan ids and sort them.
If go netlink supports this [PR](https://github.com/vishvananda/netlink/pull/863) and we could call netlink once for contiguous vlan id.

Reply:
I think it is with the aim to support config like:
{ "minID": 100, "maxID": 200 }
{ "minID": 150, "maxID": 200 }

if you use a slice directly, it will append ids 150 to 200 again.

Another thing, the config { "maxID": 120 } does not affect the code (if in 171 addresses this), but shouldn't we return an error saying that maxID was set but not minID?

Reply:
> Another thing, the config { "maxID": 120 } does not affect the code (if in 171 addresses this), but shouldn't we return an error saying that maxID was set but not minID?

That is a problem, let me think how to deal with it.
Thanks for review!

Reply:
You can add the contents of the first if as an else of the second one. You don't need to check again if item.MinID or item.MaxID are != nil again.

Reply:
The problem is we have 3 case here
1. minID == nil && maxID == nil; need to check ID
2. (minID != nil && maxID == nil) || (minID == nil && maxID != nil); fail case; L174 try to capture.
3. minID != nil && maxID != nul; success case

L157 only capture case 3, so I need to use L174 to check case 1 or 2.
If case 2, and we return error.
otherwise, we should use the condition statement in case 2 in the code

Reply:
I fix this.
I had wrong statement.
current `else if` should be correct.

Reply:
if item.MinID != nil || item.MaxID != nil {
   print error
} else {
  do work
}

this is what I meant

Reply:
So put `single vid work` in the else statement?

Reply:
```
if item.MinID != nil || item.MaxID != nil {
print error
} else {
//do work
// add single vid 
}
```

or 

```
if item.MinID != nil || item.MaxID != nil {
print error
} else {
//do work
// add vlan range from minID to maxID
}
```

Reply:
ignore my last comment, you latest fix is ok